publication-commons/src/main/java/no/unit/nva/publication/model/storage/Dao.java

```diff
@@ -83,122 +87,124 @@ public static String orgUriToOrgIdentifier(URI uri) {
      *
      * @return filtering expression string.
      */
-    public static String scanFilterExpressionForDataEntries() {
-        return "begins_with (#PK, :Resource) or "
-               + "begins_with(#PK, :Ticket) or "
-               + "begins_with(#PK, :DoiRequest) or "
-               + "begins_with(#PK, :Message)";
+    public static String scanFilterExpressionForDataEntries(Collection<KeyField> types) {
+        var currentTypes = types.isEmpty() ? List.of(KeyField.values()) : types;
```

Comment:
Could potentially pull this out as a method since it is re-used on  line 102…probably returning a stream.